docs/pinot_hadoop.rst

```diff
@@ -48,151 +48,133 @@ Create a job properties configuration file, such as one below:
+#. Prepare a schema file describing the schema of the input data. The schema needs to be in JSON format. See example later in this section.
+#. Specifically for CSV format, an optional csv config file can be provided (also in JSON format). This is used to configure parameters like the delimiter/header for the CSV file etc. A detailed description of this follows below.
 
 Run the pinot-admin command to generate the segments. The command can be invoked as follows. Options within "[ ]" are optional. For -format, the default value is AVRO.
 
-.. code-block:: none
-
-    bin/pinot-admin.sh CreateSegment -dataDir <input_data_dir> [-format [CSV/JSON/AVRO]] [-readerConfigFile <csv_config_file>] [-generatorConfigFile <generator_config_file>] -segmentName <segment_name> -schemaFile <input_schema_file> -tableName <table_name> -outDir <output_data_dir> [-overwrite]
+.. code-block:: bash
 
+   bin/pinot-admin.sh CreateSegment -dataDir <input_data_dir> [-format [CSV/JSON/AVRO]] [-readerConfigFile <csv_config_file>] [-generatorConfigFile <generator_config_file>] -segmentName <segment_name> -schemaFile <input_schema_file> -tableName <table_name> -outDir <output_data_dir> [-overwrite]
 
 To configure various parameters for CSV a config file in JSON format can be provided. This file is optional, as are each of its parameters. When not provided, default values used for these parameters are described below:
 
-#.  fileFormat: Specify one of the following. Default is EXCEL.
+#. fileFormat: Specify one of the following. Default is EXCEL.
 
- ##.  EXCEL
- ##.  MYSQL
- ##.  RFC4180
- ##.  TDF
+   #. EXCEL
+   #. MYSQL
+   #. RFC4180
+   #. TDF
 
-#.  header: If the input CSV file does not contain a header, it can be specified using this field. Note, if this is specified, then the input file is expected to not contain the header row, or else it will result in parse error. The columns in the header must be delimited by the same delimiter character as the rest of the CSV file.
-#.  delimiter: Use this to specify a delimiter character. The default value is ",".
-#.  dateFormat: If there are columns that are in date format and need to be converted into Epoch (in milliseconds), use this to specify the format. Default is "mm-dd-yyyy".
-#.  dateColumns: If there are multiple date columns, use this to list those columns.
+#. header: If the input CSV file does not contain a header, it can be specified using this field. Note, if this is specified, then the input file is expected to not contain the header row, or else it will result in parse error. The columns in the header must be delimited by the same delimiter character as the rest of the CSV file.
+#. delimiter: Use this to specify a delimiter character. The default value is ",".
+#. multiValueDelimiter: Use this to specify a delimiter character for each value in multi-valued columns. The default value is ";".
```

Comment:
shall we also mention this is only supported in `CSVRecordReader` for now? and btw, do we have a plan to support this for all kinds of RecordReaders?

Reply:
We don't need them for most record readers, because the structure of the file format already splits the values. E.g. in Json format, different values are under different column names, and multi-valued column are modeled as array.